src/main/java/net/fabricmc/loader/FabricLoader.java

```diff
@@ -197,18 +185,19 @@ public void load() {
 		if (provider == null) throw new IllegalStateException("game provider not set");
 		if (frozen) throw new IllegalStateException("Frozen - cannot load additional mods!");
 
-		try {
-			setup();
-		} catch (ModResolutionException exception) {
-			FabricGuiEntry.displayCriticalError(exception, true);
-		}
+		Set<Throwable> exceptions = setup();
+		if (!exceptions.isEmpty())
+			FabricGuiEntry.displayCriticalErrors(exceptions, true);
```

Comment:
Instead of doing this (and changing `FabricGuiEntry`), I think it would make more sense to:
1. Use an `ArrayList` of `Throwable` rather than a set (so we keep the order - since we're now swallowing exceptions and continuing I don't think we can rely on any other than the first being 100% accurate).
2. Add every exception (after the first one) as a "supressed" exception to the first one, and then pass that `FabricGuiEntry.displayCriticalError` rather than the full list. Alternatively you could create a different top-level exception to contain all of them. (But still in order)